licensedb/internal/db.go

```diff
@@ -462,7 +462,23 @@ func tfidf(freq int, docfreq int, ndocs int) float32 {
 }
 
 func (db *database) QuerySourceFile(text string) map[string]float32 {
-	// TO DO: implement this function
-	placeholder := map[string]float32{}
-	return
+	candidates := map[string]float32{}
+	append := func(others map[string]float32) {
+		for key, val := range others {
+			if candidates[key] < val {
+				candidates[key] = val
+			}
+		}
+	}
+	append(db.QueryLicenseText(string(text)))
+	if len(candidates) == 0 {
+		// TO DO: split license-comments from description-comments.
```

Comment:
As all of the files in the project are scanned for licenses I though about doing this split here for performance's sake, what do you think? Should it be done before to improve % on the output?